### ISSUE TITLE
File-configured item profile is not updated in Admin UI (only after reboot)

### ISSUE THREAD

**Issue by quensen** [openhab/openhab-core#4277]
## Context
I use only the text file based configuration, not the UI config. There, I configured things and items (in this case for the KNX and MQTT bindings).
I use my KNX button sensors (_GIRA Tastsensor 3 plus_) not (any more) to switch the power directly (using a KNX actor) but to switch a KNX Group Address ON/OFF that is a file configured thing linked to a fle configured item in openHAB.
I also have a Zigbee light bulb that can be switched ON/OFF using a _MQTT2Zigbee_ topic that is file configured as a thing in openHAB.
It all works well in general, but when I add a MQTT channel to a KNX channel to create an item while the OH service is running 'on the fly', the profile won't be shown in the UI‘s profile radio buttons (although it works), only, when I restart the OH service.

## Expected Behavior
I can manually configure the functions above using the web-based Admin UI, set the profile for the MQTT-bound channel to the `Follow `profile using the profile radio buttons, but I intend to do it via text file. I expected the system to show the updated profile in the Admin UI (the profile radio button selector).

## Current Behavior
The file configured setting is done 'somehow internally' but it is not reflected in the Admin UI.
After a restart of the OH service, it is shown correctly. The radio button reflecting the chosen profile is not default any more.

## Possible Solution
None, the workaround is the restart of the OH service.

## Steps to Reproduce (for Bugs)
1. Have two channels configured using .things files, e.g. one for KNX, another one for MQTT (but I guess the binding does not matter here):
```
	Thing device ga "KNX-Gruppenadressen" [ ]
	{
		Channels:
			Type switch : test_mqtt_schalter "Test MQTT-Schalter" [ ga = "<31/0/99" ]
```
and
```
	Thing topic ewelink_24 "Zigbee eWeLink 2.4" @"OG Büro" [ availabilityTopic = "zigbee2mqtt/ewelink_24/availability",		payloadAvailable = "online", payloadNotAvailable = "offline" ]
	{
		Channels:
			Type switch : schalter "Schalter" [ stateTopic = "zigbee2mqtt/ewelink_24/state",						commandTopic = "zigbee2mqtt/ewelink_24/set", formatBeforePublish = "{\"state\":\"%s\"}" ]

```
2. Create an item using a .items file
3. In the .items file, link the two channels with a profile while the OH services is running.
```
Switch Test_MQTT_Schalter "Test MQTT-Schalter" { channel = "knx:device:router:ga:test_mqtt_schalter", channel = "mqtt:topic:mosquitto:ewelink_24:schalter"[profile="system:follow"] }
```
4. Check item config in the Admin UI
`http://my_host:8080/settings/items/Test_MQTT_Schalter/links/edit/mqtt:topic:mosquitto:ewelink_24:schalter`

## Your Environment
Server: 
- Release = Debian GNU/Linux 12 (bookworm)
- Kernel = Linux 6.6.31+rpt-rpi-2712
- Platform = BCM43455 37.4MHz Raspberry Pi 3+-0190

Browser: MS Edge on the PC

**Comment by lolodomo** [comment]
Is your problem reproducible by just updating an existing item inside an items file by setting the follow profile ?
Or is it the combination of creating a new item and setting the follow profile ?

**Comment by lolodomo** [comment]
> The file configured setting is done 'somehow internally' but it is not reflected in the Admin UI.

Maybe a bug on Main UI ?
Did you try Ctrl+F5 or the reload app link in the about page?

**Comment by miloit** [comment]
I can confirm this issue even the health check is naming the not existing item

**Comment by quensen** [comment]
> I can confirm this issue even the health check is naming the not existing item

Hello miloit,

2 questions:
- What do you mean with the not existing item? The item exists in the Admin UI and both of the channels are listed below, but the profile radio button in the UI does not switch accordingly when I add a profile to one of the channel's.
- Maybe a bit off-topic, but there is a health check? Where?

Best regards,
Stefano


>

**Comment by quensen** [comment]
> > The file configured setting is done 'somehow internally' but it is not reflected in the Admin UI.
> 
> Maybe a bug in Main UI ? Did you try Ctrl+F5 or the reload app link in the about page?

Yes, I did. Only a restart of the OH service updated the display correctly.

**Comment by lolodomo** [comment]
I will try to reproduce an easier use case without MQTT, just adding a profile to an item in an items file

**Comment by quensen** [comment]
> I will try to reproduce an easier use case without MQTT, just adding a profile to an item in an items file

Yes, please. And please use one item but two channels form things of different bindings configured in separate things files. I have a `knx.things` file and a `mqtt.things` file with and one `knx.items` file that combines two channels into one item by taking one channel from two different things.

 I also think that the binding does not matter. I just explained the background.

**Comment by miloit** [comment]
See here

https://community.openhab.org/t/openhab-4-2-milestone-discussion/154316/176?u=milo


Maybe a different topic but the failure behavior looks similar to me


Health check was newly introduced with oh4.2M4

**Comment by lolodomo** [comment]
Not sure where you are looking at in Main UI.
A very simple case like this one:
```
Number TemperatureChambre "Température chambre [%.1f °C]" <temperature> (GSondeChambre,GTemperatureInt) [ "Measurement", "Temperature" ] {channel="rfxcom:temperaturehumidity:rfx:Chambre:temperature" [profile="offset", offset="0.6"] }
```

When I check the API /links  for this item, it returns as expected:
```
[
  {
    "editable": false,
    "channelUID": "rfxcom:temperaturehumidity:rfx:Chambre:temperature",
    "configuration": {
      "offset": "0.6",
      "profile": "offset"
    },
    "itemName": "TemperatureChambre"
  }
]
```

But in Main UI, I see that:
![image](https://github.com/user-attachments/assets/d37182a4-3837-4f25-a42c-33001067694f)
@quensen : you are looking at the same screen ?

The profile was not added after startup but was already there before I start OH.

@florian-h05 : looks like a Main UI bug, no ?

**Comment by quensen** [comment]
Salut @lolodomo ,

yes, I am looking on that display, but I was talking about two items.

Salutations,
Stefan

**Comment by lolodomo** [comment]
If I create an item in UI and link it with a profile, then the same page is showing the profile:
![image](https://github.com/user-attachments/assets/2132978d-39de-4486-8163-31c6a33dbabd)

Here is the response of the API /links:
```
[
  {
    "editable": true,
    "channelUID": "rfxcom:temperaturehumidity:rfx:Chambre:temperature",
    "configuration": {
      "offset": "0.5",
      "profile": "system:offset"
    },
    "itemName": "TemperatureTest"
  }
]
```

Interesting, profile is "system:offset" while it is just "offset" in the other case.
@florian-h05 : could it be the source of the bug ?

**Comment by lolodomo** [comment]
> yes, I am looking on that display, but I was talking about two items.

Looks like there is a bug already with a very simple use case with just one item linked to a channel with a profile.

**Comment by quensen** [comment]
_Interesting, profile is "system:offset" while it is just "offset" in the other case._

I think `system: ` is the default for the profile source. You can also create your own profiles. Then, you need the prefix.

**Comment by lolodomo** [comment]
I will try to use "system:offset" in file to see if Main UI is then able to detect it. That would confirm the bug in Main UI.

**Comment by lolodomo** [comment]
And that works:
![image](https://github.com/user-attachments/assets/568df31b-8090-487c-b4e4-0cdeffa32092)

@florian-h05 : there is a clear bug in Main UI, the profile is not detected in case it does not have a prefix.
"offset" => not detected
"system:offset" => properly detected

Do you need another issue declared in webui repo ?

@quensen : as a workaround, please add the prefix "system:" to all your profile definitions and let us know if your issue is solved.

**Comment by J-N-K** [comment]
We could also add `system:` during  link creation when no prefix is present. IMO all these "implicit" prefixes should be added at the first place possible.

**Comment by lolodomo** [comment]
> We could also add `system:` during link creation when no prefix is present. IMO all these "implicit" prefixes should be added at the first place possible.

In our current case, the link is coming from an items file and the prefix is not present.

**Comment by lolodomo** [comment]
> We could also add `system:` during link creation when no prefix is present.

Maybe you mean when loading the file ?

**Comment by J-N-K** [comment]
Exactly, in `GenericItemChannelLinkProvider` when the link is created.

**Comment by florian-h05** [comment]
> We could also add system: during link creation when no prefix is present. IMO all these "implicit" prefixes should be added at the first place possible.

Agreed - I do not consider it as an UI bug that UI does not display the profile, because UI gets all available profile types from REST (e.g. `/rest/profile-types?channelTypeUID=astro:end`) and the response is like this:

```json
[
    {
        "uid": "system:default",
        "label": "Default",
        "kind": "STATE",
        "supportedItemTypes": []
    },
    {
        "uid": "system:follow",
        "label": "Follow",
        "kind": "STATE",
        "supportedItemTypes": []
    },
    {
        "uid": "transform:JSONPATH",
        "label": "JSONPATH",
        "kind": "STATE",
        "supportedItemTypes": []
    },
    {
   ...
]
```

So I would expect to get these profile ids returned for the selected profile and not to have to make any assumptions in the UI.

**Comment by lolodomo** [comment]
I am going to fix loading of file, as suggested by @J-N-K , to add system prefix when missing in user file.

**Comment by lolodomo** [comment]
While I was looking at the code for system profiles, I might have found a bug in class SystemProfileFactory.
The method  createProfile handles 24 profiles but the methods getProfileTypes and getSupportedProfileTypeUIDs only consider 21 types, those defined respectively in constants SUPPORTED_PROFILE_TYPES and SUPPORTED_PROFILE_TYPE_UIDS.
The missing profiles in these 2 constants are: BUTTON_TOGGLE_SWITCH, BUTTON_TOGGLE_PLAYER and BUTTON_TOGGLE_ROLLERSHUTTER.
@J-N-K : do you confirm ?

**Comment by J-N-K** [comment]
They seem to be missing, correct.

**Comment by lolodomo** [comment]
> They seem to be missing, correct.

I tried to add them but in fact I have no precise idea how each ProfileType should be defined.

**Comment by MarkusElsing** [comment]
Hi, the main issue seems to be fixed in 4.2.2, but I noticed that the system health check does not update if the .items files is updated e.g. for the MQTT channels of a node. After the .items file reload the link is for the items are set correctly, but they are still listed in the health check as broken links to non-existing channels.
